### PR TITLE
Remove Route from RoutingKit

### PR DESCRIPTION
**Pull request by MrLotU**
This PR removes the `Route` class from `RoutingKit`. Reasoning for this is that `RoutingKit` should be purely focussed on speed and performance. The use of `Route` obsolete here, since it's only ever used internally or for registration. Because of that we can remove it. Extended discussion can be found here: https://canary.discordapp.com/channels/431917998102675485/519613337638797315/653612293443158045

### Checklist

- [x] Circle CI is passing (code compiles and passes tests).
- [ ] There are no breaking changes to public API.
- [x] New test cases have been added where appropriate.
- [x] All new code has been commented with doc blocks `///`.

### CODE REVIEW

**Comment by MrLotU** [comment]
@vapor-bot test performance

**Comment by twof** [comment]
@vapor-bot test performance

**Comment by MrLotU** [comment]
@vapor-bot test performance

**Comment by MrLotU** [comment]
@twof Circle reports `webhook returned http status 404, connecting to https://bot-gh.vapor.codes/circle/result`. Could you see what's happening? I don't think anything changed 🤔

**Comment by twof** [comment]
@MrLotU Yeah I'll check that out later. Can you summarize the reasons for removing `Route`? I'd prefer having the reasoning here incase we ever lose discord chat history.

**Comment by MrLotU** [comment]
I updated the issue description with some more context 👍

**Comment by twof** [comment]
Thanks!

File: Sources/RoutingKit/Router.swift [hunk 1]
```diff
@@ -5,10 +5,13 @@ public protocol Router {
...
-    /// Registers a new `Route` to the `Router`.
+    /// Registers a new `Output` to the `Router` at a given path.
     ///
-    /// Extraneous information such as `userInfo` may be discarded.
-    mutating func register(route: Route<Output>)
+    ///  - parameters:
+    ///     - output: Output to register.
+    ///     - path: Path to register output at.
+    ///
+    mutating func register(_ output: Output, at path: [PathComponent])
```
Comment (tanner0101) [inline Sources/RoutingKit/Router.swift]: I think I'd prefer `register(path:output:)` here since it better aligns with the other method on this protocol `route(path:parameters:)`

File: Tests/LinuxMain.swift [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import XCTest
```
Comment (tanner0101) [inline Tests/LinuxMain.swift]: Why is this back?

File: circle.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+jobs:
+  linux-performance:
+    docker:
+      - image: swift:5.0-bionic
+    steps:
+      - checkout
+      - run:
+          name: swift test
+          command: swift test -c release --filter "RoutingKitTests.RouterPerformanceTests"
```
Comment (tanner0101) [inline circle.yml]: we should be able to do `swift test --enable-test-discovery` here if you do `swift:5.1-bionic`

File: Tests/RoutingKitTests/XCTestManifests.swift [hunk 4]
```diff
@@ -0,0 +1,44 @@
+#if !canImport(ObjectiveC)
```
Comment (tanner0101) [inline Tests/RoutingKitTests/XCTestManifests.swift]: rm rm rm rm rm!!

File: Sources/RoutingKit/Router.swift (hunk 1 above)
Comment (MrLotU) [reply Sources/RoutingKit/Router.swift]: I'd be fine with that, but you register output at a path, right? Or would you then rather have `register(path, with: output)`.

File: Tests/LinuxMain.swift (hunk 2 above)
Comment (MrLotU) [reply Tests/LinuxMain.swift]: Because I forgot about test discovery :) I'll remove it.

File: circle.yml (hunk 3 above)
Comment (MrLotU) [reply circle.yml]: Will update

File: Sources/RoutingKit/Router.swift (hunk 1 above)
Comment (tanner0101) [reply Sources/RoutingKit/Router.swift]: I think just `register(path:output:)`. It's very much a nit though, approving.

**Comment by tanner0101** [comment]
This is a breaking change so we'll leave on master until we're ready for beta.3
